RealmSwift/Tests/RealmCollectionTypeTests.swift

```diff
@@ -689,7 +689,7 @@ class ResultsWithCustomInitializerTest: TestCase {
         let expected = Array(collection.map { $0.stringCol })
         let actual = collection.value(forKey: "stringCol") as! [String]!
         XCTAssertEqual(expected, actual!)
-        XCTAssertEqual(collection.map { $0 }, collection.value(forKey: "self") as! [CTTStringObjectWithLink])
+        assertEqualObjectCollections(collection.map { $0 }, collection.value(forKey: "self") as! [SwiftCustomInitializerObject])
```

Comment:
How was this test passing before?

Unrelated to this PR, but we should rename `ResultsWithCustomInitializerTest` to `ResultsWithCustomInitializerTests` to match our test case naming convention.